[WebGPU] api,validation,encoding,* is failing

mwyrzykowski opened the pull request:
#### e15eaf54e016dd9309122ea662c664a5d8ae10a6
<pre>
[WebGPU] api,validation,encoding,* is failing
<a href="https://bugs.webkit.org/show_bug.cgi?id=267285">https://bugs.webkit.org/show_bug.cgi?id=267285</a>
&lt;radar://120727214&gt;

Reviewed by Dan Glastonbury.

Add validation and passing expectations for validation,encoding,*
and its 24 test suites.

Also disable timestamps query as the counter sample buffers were sometimes returning
nil and the error message from the debug layer was not useful. Re-enabling is tracked
by <a href="https://bugs.webkit.org/show_bug.cgi?id=268244">https://bugs.webkit.org/show_bug.cgi?id=268244</a>

* LayoutTests/http/tests/webgpu/common/util/util.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginComputePass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginRenderPass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginRenderPass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/clearBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/clearBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/compute_pass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/compute_pass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyBufferToBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyBufferToBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/debug-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/debug.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/index_access-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/index_access.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/draw.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/dynamic_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/dynamic_state.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/indirect_draw-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/indirect_draw.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/render.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setIndexBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setIndexBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setPipeline-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setPipeline.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setVertexBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setVertexBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/state_tracking-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/state_tracking.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render_pass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render_pass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/setBindGroup-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/setBindGroup.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/createRenderBundleEncoder-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/createRenderBundleEncoder.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_open_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_open_state.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_state.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/begin_end-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/begin_end.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/common.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/general-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/general.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/pipeline_statistics-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/render_bundle-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/render_bundle.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/validation_test.js:
* LayoutTests/http/tests/webgpu/webgpu/capability_info.js:
* LayoutTests/http/tests/webgpu/webgpu/util/device_pool.js:
* Source/WebCore/Modules/WebGPU/GPUComputePassEncoder.cpp:
(WebCore::GPUComputePassEncoder::setBindGroup):
* Source/WebCore/Modules/WebGPU/GPUComputePassEncoder.h:
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoder.cpp:
(WebCore::GPURenderBundleEncoder::setVertexBuffer):
(WebCore::GPURenderBundleEncoder::setBindGroup):
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoder.h:
* Source/WebCore/Modules/WebGPU/GPURenderEncoderBase.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassDescriptor.h:
(WebCore::GPURenderPassDescriptor::convertToBacking const):
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.cpp:
(WebCore::GPURenderPassEncoder::setVertexBuffer):
(WebCore::GPURenderPassEncoder::setBindGroup):
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.h:
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUCommandEncoderImpl.cpp:
(WebCore::WebGPU::CommandEncoderImpl::beginRenderPass):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderBundleEncoderImpl.cpp:
(WebCore::WebGPU::RenderBundleEncoderImpl::setVertexBuffer):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderBundleEncoderImpl.h:
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.cpp:
(WebCore::WebGPU::RenderPassEncoderImpl::setVertexBuffer):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderBundleEncoder.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassDescriptor.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassEncoder.h:
* Source/WebGPU/WebGPU/BindGroup.h:
(WebGPU::BindGroup::create):
(WebGPU::BindGroup::isValid const): Deleted.
* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::Device::createBindGroup):
(WebGPU::BindGroup::isValid const):
(WebGPU::BindGroup::bindGroupLayout const):
(WebGPU::BindGroup::BindGroup):
(WebGPU::BindGroup::dynamicBuffer const):
* Source/WebGPU/WebGPU/BindGroupLayout.h:
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(WebGPU::BindGroupLayout::dynamicBufferCount const):
(WebGPU::BindGroupLayout::validateDynamicOffsets const):
(WebGPU::isEqual):
(WebGPU::equalBindingEntries):
(WebGPU::equalEntries):
(WebGPU::BindGroupLayout::errorValidatingBindGroupCompatibility const):
* Source/WebGPU/WebGPU/Buffer.h:
(WebGPU::Buffer::isValid const): Deleted.
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::storageMode):
(WebGPU::Buffer::setCommandEncoder const):
(WebGPU::Buffer::destroy):
(WebGPU::Buffer::unmap):
(WebGPU::Buffer::isValid const):
(WebGPU::Buffer::recomputeMaxIndexValues const):
(WebGPU::Buffer::maxIndex const):
* Source/WebGPU/WebGPU/CommandBuffer.h:
* Source/WebGPU/WebGPU/CommandBuffer.mm:
(WebGPU::CommandBuffer::makeInvalid):
(WebGPU::CommandBuffer::lastError const):
* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::~CommandEncoder):
(WebGPU::CommandEncoder::ensureBlitCommandEncoder):
(WebGPU::CommandEncoder::finalizeBlitCommandEncoder):
(WebGPU::errorValidatingTimestampWrites):
(WebGPU::CommandEncoder::errorValidatingComputePassDescriptor const):
(WebGPU::CommandEncoder::beginComputePass):
(WebGPU::CommandEncoder::setExistingEncoder):
(WebGPU::CommandEncoder::endEncoding):
(WebGPU::CommandEncoder::errorValidatingRenderPassDescriptor const):
(WebGPU::createSimplePso):
(WebGPU::CommandEncoder::runClearEncoder):
(WebGPU::CommandEncoder::beginRenderPass):
(WebGPU::CommandEncoder::errorValidatingCopyBufferToBuffer):
(WebGPU::CommandEncoder::copyBufferToBuffer):
(WebGPU::CommandEncoder::copyBufferToTexture):
(WebGPU::CommandEncoder::makeInvalid):
(WebGPU::CommandEncoder::makeSubmitInvalid):
(WebGPU::CommandEncoder::copyTextureToBuffer):
(WebGPU::validateCopyTextureToTexture):
(WebGPU::CommandEncoder::copyTextureToTexture):
(WebGPU::CommandEncoder::clearBuffer):
(WebGPU::CommandEncoder::setLastError):
(WebGPU::CommandEncoder::finish):
(WebGPU::CommandEncoder::insertDebugMarker):
(WebGPU::CommandEncoder::popDebugGroup):
(WebGPU::CommandEncoder::pushDebugGroup):
(WebGPU::validateResolveQuerySet):
(WebGPU::CommandEncoder::resolveQuerySet):
(WebGPU::CommandEncoder::writeTimestamp):
(WebGPU::CommandEncoder::lock):
(WebGPU::CommandEncoder::isLocked const):
(WebGPU::CommandEncoder::isFinished const):
(WebGPU::validateTimestampWrites): Deleted.
(WebGPU::CommandEncoder::validateComputePassDescriptor const): Deleted.
(WebGPU::CommandEncoder::validateRenderPassDescriptor const): Deleted.
* Source/WebGPU/WebGPU/CommandsMixin.h:
* Source/WebGPU/WebGPU/CommandsMixin.mm:
(WebGPU::CommandsMixin::encoderStateName const):
(WebGPU::CommandsMixin::computedSizeOverflows):
* Source/WebGPU/WebGPU/ComputePassEncoder.h:
(WebGPU::ComputePassEncoder::createInvalid):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::ComputePassEncoder):
(WebGPU::ComputePassEncoder::~ComputePassEncoder):
(WebGPU::ComputePassEncoder::executePreDispatchCommands):
(WebGPU::ComputePassEncoder::dispatch):
(WebGPU::ComputePassEncoder::runPredispatchIndirectCallValidation):
(WebGPU::ComputePassEncoder::dispatchIndirect): Deleted.
(WebGPU::ComputePassEncoder::endPass): Deleted.
(WebGPU::ComputePassEncoder::insertDebugMarker): Deleted.
(WebGPU::ComputePassEncoder::validatePopDebugGroup const): Deleted.
(WebGPU::ComputePassEncoder::makeInvalid): Deleted.
(WebGPU::ComputePassEncoder::popDebugGroup): Deleted.
(WebGPU::ComputePassEncoder::pushDebugGroup): Deleted.
(WebGPU::setCommandEncoder): Deleted.
(WebGPU::ComputePassEncoder::setBindGroup): Deleted.
(WebGPU::ComputePassEncoder::setPipeline): Deleted.
(WebGPU::ComputePassEncoder::setLabel): Deleted.
(wgpuComputePassEncoderReference): Deleted.
(wgpuComputePassEncoderRelease): Deleted.
(wgpuComputePassEncoderDispatchWorkgroups): Deleted.
(wgpuComputePassEncoderDispatchWorkgroupsIndirect): Deleted.
(wgpuComputePassEncoderEnd): Deleted.
(wgpuComputePassEncoderInsertDebugMarker): Deleted.
(wgpuComputePassEncoderPopDebugGroup): Deleted.
(wgpuComputePassEncoderPushDebugGroup): Deleted.
(wgpuComputePassEncoderSetBindGroup): Deleted.
(wgpuComputePassEncoderSetPipeline): Deleted.
(wgpuComputePassEncoderSetLabel): Deleted.
* Source/WebGPU/WebGPU/Device.h:
* Source/WebGPU/WebGPU/Device.mm:
(WebGPU::Device::Device):
(WebGPU::Device::placeholderBuffer const):
(WebGPU::Device::placeholderTexture const):
* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::baseCapabilities):
(WebGPU::baseFeatures):
(WebGPU::mergeBaseCapabilities):
* Source/WebGPU/WebGPU/PipelineLayout.h:
* Source/WebGPU/WebGPU/PipelineLayout.mm:
(WebGPU::PipelineLayout::errorValidatingBindGroupCompatibility const):
* Source/WebGPU/WebGPU/QuerySet.h:
* Source/WebGPU/WebGPU/QuerySet.mm:
(WebGPU::Device::createQuerySet):
(WebGPU::QuerySet::QuerySet):
(WebGPU::QuerySet::isValid const):
(WebGPU::QuerySet::isDestroyed const):
(WebGPU::QuerySet::destroy):
(WebGPU::QuerySet::setCommandEncoder const):
* Source/WebGPU/WebGPU/Queue.h:
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::Queue::Queue):
(WebGPU::Queue::encoderForBuffer const):
(WebGPU::Queue::setEncoderForBuffer):
(WebGPU::Queue::commandBufferWithDescriptor):
(WebGPU::Queue::errorValidatingSubmit const):
(WebGPU::Queue::commitMTLCommandBuffer):
(WebGPU::Queue::submit):
(WebGPU::Queue::writeBuffer):
(WebGPU::Queue::writeTexture):
(WebGPU::Queue::validateSubmit const): Deleted.
* Source/WebGPU/WebGPU/RenderBundle.h:
(WebGPU::RenderBundle::create):
(WebGPU::RenderBundle::createInvalid):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(WebGPU::RenderBundle::RenderBundle):
(WebGPU::RenderBundle::drawCount const):
(WebGPU::RenderBundle::lastError const):
* Source/WebGPU/WebGPU/RenderBundleEncoder.h:
(WebGPU::RenderBundleEncoder::createInvalid):
(WebGPU::RenderBundleEncoder::device const):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::RenderBundleEncoder::returnIfEncodingIsFinished):
(WebGPU::Device::createRenderBundleEncoder):
(WebGPU::RenderBundleEncoder::RenderBundleEncoder):
(WebGPU::RenderBundleEncoder::makeInvalid):
(WebGPU::RenderBundleEncoder::addResource):
(WebGPU::RenderBundleEncoder::executePreDrawCommands):
(WebGPU::RenderBundleEncoder::draw):
(WebGPU::RenderBundleEncoder::maxVertexBufferIndex const):
(WebGPU::RenderBundleEncoder::maxBindGroupIndex const):
(WebGPU::RenderBundleEncoder::errorValidatingDraw const):
(WebGPU::RenderBundleEncoder::errorValidatingDrawIndexed const):
(WebGPU::RenderBundleEncoder::runIndexBufferValidation):
(WebGPU::RenderBundleEncoder::runVertexBufferValidation):
(WebGPU::RenderBundleEncoder::drawIndexed):
(WebGPU::RenderBundleEncoder::drawIndexedIndirect):
(WebGPU::RenderBundleEncoder::drawIndirect):
(WebGPU::RenderBundleEncoder::endCurrentICB):
(WebGPU::RenderBundleEncoder::validToEncodeCommand const):
(WebGPU::RenderBundleEncoder::finish):
(WebGPU::RenderBundleEncoder::replayCommands):
(WebGPU::RenderBundleEncoder::insertDebugMarker):
(WebGPU::RenderBundleEncoder::popDebugGroup):
(WebGPU::RenderBundleEncoder::pushDebugGroup):
(WebGPU::RenderBundleEncoder::setBindGroup):
(WebGPU::RenderBundleEncoder::setIndexBuffer):
(WebGPU::RenderBundleEncoder::recordCommand):
(WebGPU::RenderBundleEncoder::setPipeline):
(WebGPU::RenderBundleEncoder::setVertexBuffer):
(wgpuRenderBundleEncoderSetVertexBuffer):
* Source/WebGPU/WebGPU/RenderPassEncoder.h:
(WebGPU::RenderPassEncoder::create):
(WebGPU::RenderPassEncoder::createInvalid):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::RenderPassEncoder):
(WebGPU::RenderPassEncoder::~RenderPassEncoder):
(WebGPU::RenderPassEncoder::beginOcclusionQuery):
(WebGPU::RenderPassEncoder::endOcclusionQuery):
(WebGPU::RenderPassEncoder::addResourceToActiveResources):
(WebGPU::RenderPassEncoder::runIndexBufferValidation):
(WebGPU::RenderPassEncoder::runVertexBufferValidation):
(WebGPU::RenderPassEncoder::maxVertexBufferIndex const):
(WebGPU::RenderPassEncoder::maxBindGroupIndex const):
(WebGPU::RenderPassEncoder::errorValidatingAndBindingBuffers):
(WebGPU::RenderPassEncoder::errorValidatingDrawIndexed const):
(WebGPU::RenderPassEncoder::incrementDrawCount):
(WebGPU::RenderPassEncoder::issuedDrawCall const):
(WebGPU::RenderPassEncoder::executePreDrawCommands):
(WebGPU::RenderPassEncoder::draw):
(WebGPU::RenderPassEncoder::drawIndexed):
(WebGPU::RenderPassEncoder::drawIndexedIndirect):
(WebGPU::RenderPassEncoder::drawIndirect):
(WebGPU::RenderPassEncoder::endPass):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::insertDebugMarker):
(WebGPU::RenderPassEncoder::validatePopDebugGroup const):
(WebGPU::RenderPassEncoder::makeInvalid):
(WebGPU::RenderPassEncoder::popDebugGroup):
(WebGPU::RenderPassEncoder::setBindGroup):
(WebGPU::RenderPassEncoder::setBlendConstant):
(WebGPU::RenderPassEncoder::setIndexBuffer):
(WebGPU::RenderPassEncoder::setPipeline):
(WebGPU::RenderPassEncoder::setScissorRect):
(WebGPU::RenderPassEncoder::setStencilReference):
(WebGPU::RenderPassEncoder::setVertexBuffer):
(WebGPU::RenderPassEncoder::setViewport):
(wgpuRenderPassEncoderSetVertexBuffer):
(WebGPU::RenderPassEncoder::validateBindGroups): Deleted.
* Source/WebGPU/WebGPU/RenderPipeline.h:
(WebGPU::RenderPipeline::create):
(WebGPU::RenderPipeline::requiredBufferIndices const):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::createVertexDescriptor):
(WebGPU::Device::createRenderPipeline):
(WebGPU::RenderPipeline::RenderPipeline):
(WebGPU::RenderPipeline::primitiveTopology const):
(WebGPU::RenderPipeline::stripIndexFormat const):
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::handleShaderSuccessOrFailure):
* Source/WebGPU/WebGPU/Texture.h:
(WebGPU::Texture::isValid const): Deleted.
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::destroy):
(WebGPU::Texture::validateImageCopyTexture):
(WebGPU::Texture::isValid const):
* Source/WebGPU/WebGPU/TextureView.h:
(WebGPU::TextureView::isValid const): Deleted.
* Source/WebGPU/WebGPU/TextureView.mm:
(WebGPU::TextureView::isValid const):
(WebGPU::TextureView::destroy):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderBundleEncoder.cpp:
(WebKit::RemoteRenderBundleEncoder::setVertexBuffer):
(WebKit::RemoteRenderBundleEncoder::unsetVertexBuffer):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderBundleEncoder.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderBundleEncoder.messages.in:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderPassEncoder.cpp:
(WebKit::RemoteRenderPassEncoder::setVertexBuffer):
(WebKit::RemoteRenderPassEncoder::unsetVertexBuffer):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderPassEncoder.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderPassEncoder.messages.in:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDescriptor.cpp:
(WebKit::WebGPU::ConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::ConvertFromBackingContext::convertFromBacking):
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDescriptor.h:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDescriptor.serialization.in:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderBundleEncoderProxy::setVertexBuffer):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderPassEncoderProxy::setVertexBuffer):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.h:

Canonical link: <a href="https://commits.webkit.org/273966@main">https://commits.webkit.org/273966@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b5905367d5818b3cf4833bc44c07daac8ac71356

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/37252 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/16153 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/39548 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/39814 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/33224 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/38555 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/18733 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/13297 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/31701 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/37816 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/13633 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/32761 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/11861 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/11857 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/33408 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/41074 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/33736 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/33840 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/37757 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/12198 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/9985 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/35900 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/13809 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8428 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/12549 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/13068 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg commented on Source/WebGPU/WebGPU/Buffer.mm:
```diff
@@ -85,7 +85,7 @@ static MTLStorageMode storageMode(bool deviceHasUnifiedMemory, WGPUBufferUsageFl
     if (deviceHasUnifiedMemory)
         return MTLStorageModeShared;
-    if ((usage & WGPUBufferUsage_MapRead) || (usage & WGPUBufferUsage_MapWrite))
+    if ((usage & WGPUBufferUsage_MapRead) || (usage & WGPUBufferUsage_MapWrite) || (usage & WGPUBufferUsage_Index))
```
something like:
```
if ((usage & (WGPUBufferUsage_MapRead | WGPUBufferUsage_MapWrite | WGPUBufferUsage_Index)) {
    ...
}
```
Would be easier for me to read.

mwyrzykowski replied:
Good idea, I took that suggestion

djg commented on Source/WebGPU/WebGPU/Buffer.mm:
```diff
@@ -361,11 +369,41 @@ static size_t computeRangeSize(uint64_t size, size_t offset)
+    m_max32BitIndex = 0;
+    uint8_t* bufferEnd = bufferPtr + lengthInBytes;
+    for (; (bufferPtr += sizeof(uint32_t)) <= bufferEnd; bufferPtr += sizeof(uint32_t)) {
+        m_max32BitIndex = std::max(*reinterpret_cast<uint32_t*>(bufferPtr), m_max32BitIndex);
```
is `bufferPtr` guaranteed to be correctly aligned to avoid misaligned loads? (Maybe that's not an issue on AS which is think behaves like x86 to make porting easier)

mwyrzykowski replied:
bufferPtr is the contents pointer of a MTLBuffer instance, so this is guaranteed to be 16 byte aligned

djg reviewed (approved):
The size of this review deserves a :ship:

mwyrzykowski committed:
[WebGPU] api,validation,encoding,* is failing
https://bugs.webkit.org/show_bug.cgi?id=267285
<radar://120727214>

Reviewed by Dan Glastonbury.

Add validation and passing expectations for validation,encoding,*
and its 24 test suites.

Also disable timestamps query as the counter sample buffers were sometimes returning
nil and the error message from the debug layer was not useful. Re-enabling is tracked
by https://bugs.webkit.org/show_bug.cgi?id=268244

* LayoutTests/http/tests/webgpu/common/util/util.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginComputePass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginRenderPass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/beginRenderPass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/clearBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/clearBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/compute_pass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/compute_pass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyBufferToBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyBufferToBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/copyTextureToTexture.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/debug-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/debug.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/index_access-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/index_access.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/draw.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/dynamic_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/dynamic_state.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/indirect_draw-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/indirect_draw.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/render.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setIndexBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setIndexBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setPipeline-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setPipeline.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setVertexBuffer-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/setVertexBuffer.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/state_tracking-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render/state_tracking.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render_pass-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/render_pass.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/setBindGroup-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/cmds/setBindGroup.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/createRenderBundleEncoder-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/createRenderBundleEncoder.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_open_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_open_state.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_state-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/encoder_state.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/programmable/pipeline_bind_group_compat.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/begin_end-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/begin_end.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/common.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/general-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/general.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/pipeline_statistics-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/queries/resolveQuerySet.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/render_bundle-expected.txt:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/encoding/render_bundle.spec.js:
* LayoutTests/http/tests/webgpu/webgpu/api/validation/validation_test.js:
* LayoutTests/http/tests/webgpu/webgpu/capability_info.js:
* LayoutTests/http/tests/webgpu/webgpu/util/device_pool.js:
* Source/WebCore/Modules/WebGPU/GPUComputePassEncoder.cpp:
(WebCore::GPUComputePassEncoder::setBindGroup):
* Source/WebCore/Modules/WebGPU/GPUComputePassEncoder.h:
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoder.cpp:
(WebCore::GPURenderBundleEncoder::setVertexBuffer):
(WebCore::GPURenderBundleEncoder::setBindGroup):
* Source/WebCore/Modules/WebGPU/GPURenderBundleEncoder.h:
* Source/WebCore/Modules/WebGPU/GPURenderEncoderBase.idl:
* Source/WebCore/Modules/WebGPU/GPURenderPassDescriptor.h:
(WebCore::GPURenderPassDescriptor::convertToBacking const):
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.cpp:
(WebCore::GPURenderPassEncoder::setVertexBuffer):
(WebCore::GPURenderPassEncoder::setBindGroup):
* Source/WebCore/Modules/WebGPU/GPURenderPassEncoder.h:
* Source/WebCore/Modules/WebGPU/Implementation/WebGPUCommandEncoderImpl.cpp:
(WebCore::WebGPU::CommandEncoderImpl::beginRenderPass):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderBundleEncoderImpl.cpp:
(WebCore::WebGPU::RenderBundleEncoderImpl::setVertexBuffer):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderBundleEncoderImpl.h:
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.cpp:
(WebCore::WebGPU::RenderPassEncoderImpl::setVertexBuffer):
* Source/WebCore/Modules/WebGPU/Implementation/WebGPURenderPassEncoderImpl.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderBundleEncoder.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassDescriptor.h:
* Source/WebCore/Modules/WebGPU/InternalAPI/WebGPURenderPassEncoder.h:
* Source/WebGPU/WebGPU/BindGroup.h:
(WebGPU::BindGroup::create):
(WebGPU::BindGroup::isValid const): Deleted.
* Source/WebGPU/WebGPU/BindGroup.mm:
(WebGPU::Device::createBindGroup):
(WebGPU::BindGroup::isValid const):
(WebGPU::BindGroup::bindGroupLayout const):
(WebGPU::BindGroup::BindGroup):
(WebGPU::BindGroup::dynamicBuffer const):
* Source/WebGPU/WebGPU/BindGroupLayout.h:
* Source/WebGPU/WebGPU/BindGroupLayout.mm:
(WebGPU::BindGroupLayout::dynamicBufferCount const):
(WebGPU::BindGroupLayout::validateDynamicOffsets const):
(WebGPU::isEqual):
(WebGPU::equalBindingEntries):
(WebGPU::equalEntries):
(WebGPU::BindGroupLayout::errorValidatingBindGroupCompatibility const):
* Source/WebGPU/WebGPU/Buffer.h:
(WebGPU::Buffer::isValid const): Deleted.
* Source/WebGPU/WebGPU/Buffer.mm:
(WebGPU::storageMode):
(WebGPU::Buffer::setCommandEncoder const):
(WebGPU::Buffer::destroy):
(WebGPU::Buffer::unmap):
(WebGPU::Buffer::isValid const):
(WebGPU::Buffer::recomputeMaxIndexValues const):
(WebGPU::Buffer::maxIndex const):
* Source/WebGPU/WebGPU/CommandBuffer.h:
* Source/WebGPU/WebGPU/CommandBuffer.mm:
(WebGPU::CommandBuffer::makeInvalid):
(WebGPU::CommandBuffer::lastError const):
* Source/WebGPU/WebGPU/CommandEncoder.h:
* Source/WebGPU/WebGPU/CommandEncoder.mm:
(WebGPU::CommandEncoder::~CommandEncoder):
(WebGPU::CommandEncoder::ensureBlitCommandEncoder):
(WebGPU::CommandEncoder::finalizeBlitCommandEncoder):
(WebGPU::errorValidatingTimestampWrites):
(WebGPU::CommandEncoder::errorValidatingComputePassDescriptor const):
(WebGPU::CommandEncoder::beginComputePass):
(WebGPU::CommandEncoder::setExistingEncoder):
(WebGPU::CommandEncoder::endEncoding):
(WebGPU::CommandEncoder::errorValidatingRenderPassDescriptor const):
(WebGPU::createSimplePso):
(WebGPU::CommandEncoder::runClearEncoder):
(WebGPU::CommandEncoder::beginRenderPass):
(WebGPU::CommandEncoder::errorValidatingCopyBufferToBuffer):
(WebGPU::CommandEncoder::copyBufferToBuffer):
(WebGPU::CommandEncoder::copyBufferToTexture):
(WebGPU::CommandEncoder::makeInvalid):
(WebGPU::CommandEncoder::makeSubmitInvalid):
(WebGPU::CommandEncoder::copyTextureToBuffer):
(WebGPU::validateCopyTextureToTexture):
(WebGPU::CommandEncoder::copyTextureToTexture):
(WebGPU::CommandEncoder::clearBuffer):
(WebGPU::CommandEncoder::setLastError):
(WebGPU::CommandEncoder::finish):
(WebGPU::CommandEncoder::insertDebugMarker):
(WebGPU::CommandEncoder::popDebugGroup):
(WebGPU::CommandEncoder::pushDebugGroup):
(WebGPU::validateResolveQuerySet):
(WebGPU::CommandEncoder::resolveQuerySet):
(WebGPU::CommandEncoder::writeTimestamp):
(WebGPU::CommandEncoder::lock):
(WebGPU::CommandEncoder::isLocked const):
(WebGPU::CommandEncoder::isFinished const):
(WebGPU::validateTimestampWrites): Deleted.
(WebGPU::CommandEncoder::validateComputePassDescriptor const): Deleted.
(WebGPU::CommandEncoder::validateRenderPassDescriptor const): Deleted.
* Source/WebGPU/WebGPU/CommandsMixin.h:
* Source/WebGPU/WebGPU/CommandsMixin.mm:
(WebGPU::CommandsMixin::encoderStateName const):
(WebGPU::CommandsMixin::computedSizeOverflows):
* Source/WebGPU/WebGPU/ComputePassEncoder.h:
(WebGPU::ComputePassEncoder::createInvalid):
* Source/WebGPU/WebGPU/ComputePassEncoder.mm:
(WebGPU::ComputePassEncoder::ComputePassEncoder):
(WebGPU::ComputePassEncoder::~ComputePassEncoder):
(WebGPU::ComputePassEncoder::executePreDispatchCommands):
(WebGPU::ComputePassEncoder::dispatch):
(WebGPU::ComputePassEncoder::runPredispatchIndirectCallValidation):
(WebGPU::ComputePassEncoder::dispatchIndirect): Deleted.
(WebGPU::ComputePassEncoder::endPass): Deleted.
(WebGPU::ComputePassEncoder::insertDebugMarker): Deleted.
(WebGPU::ComputePassEncoder::validatePopDebugGroup const): Deleted.
(WebGPU::ComputePassEncoder::makeInvalid): Deleted.
(WebGPU::ComputePassEncoder::popDebugGroup): Deleted.
(WebGPU::ComputePassEncoder::pushDebugGroup): Deleted.
(WebGPU::setCommandEncoder): Deleted.
(WebGPU::ComputePassEncoder::setBindGroup): Deleted.
(WebGPU::ComputePassEncoder::setPipeline): Deleted.
(WebGPU::ComputePassEncoder::setLabel): Deleted.
(wgpuComputePassEncoderReference): Deleted.
(wgpuComputePassEncoderRelease): Deleted.
(wgpuComputePassEncoderDispatchWorkgroups): Deleted.
(wgpuComputePassEncoderDispatchWorkgroupsIndirect): Deleted.
(wgpuComputePassEncoderEnd): Deleted.
(wgpuComputePassEncoderInsertDebugMarker): Deleted.
(wgpuComputePassEncoderPopDebugGroup): Deleted.
(wgpuComputePassEncoderPushDebugGroup): Deleted.
(wgpuComputePassEncoderSetBindGroup): Deleted.
(wgpuComputePassEncoderSetPipeline): Deleted.
(wgpuComputePassEncoderSetLabel): Deleted.
* Source/WebGPU/WebGPU/Device.h:
* Source/WebGPU/WebGPU/Device.mm:
(WebGPU::Device::Device):
(WebGPU::Device::placeholderBuffer const):
(WebGPU::Device::placeholderTexture const):
* Source/WebGPU/WebGPU/HardwareCapabilities.mm:
(WebGPU::baseCapabilities):
(WebGPU::baseFeatures):
(WebGPU::mergeBaseCapabilities):
* Source/WebGPU/WebGPU/PipelineLayout.h:
* Source/WebGPU/WebGPU/PipelineLayout.mm:
(WebGPU::PipelineLayout::errorValidatingBindGroupCompatibility const):
* Source/WebGPU/WebGPU/QuerySet.h:
* Source/WebGPU/WebGPU/QuerySet.mm:
(WebGPU::Device::createQuerySet):
(WebGPU::QuerySet::QuerySet):
(WebGPU::QuerySet::isValid const):
(WebGPU::QuerySet::isDestroyed const):
(WebGPU::QuerySet::destroy):
(WebGPU::QuerySet::setCommandEncoder const):
* Source/WebGPU/WebGPU/Queue.h:
* Source/WebGPU/WebGPU/Queue.mm:
(WebGPU::Queue::Queue):
(WebGPU::Queue::encoderForBuffer const):
(WebGPU::Queue::setEncoderForBuffer):
(WebGPU::Queue::commandBufferWithDescriptor):
(WebGPU::Queue::errorValidatingSubmit const):
(WebGPU::Queue::commitMTLCommandBuffer):
(WebGPU::Queue::submit):
(WebGPU::Queue::writeBuffer):
(WebGPU::Queue::writeTexture):
(WebGPU::Queue::validateSubmit const): Deleted.
* Source/WebGPU/WebGPU/RenderBundle.h:
(WebGPU::RenderBundle::create):
(WebGPU::RenderBundle::createInvalid):
* Source/WebGPU/WebGPU/RenderBundle.mm:
(WebGPU::RenderBundle::RenderBundle):
(WebGPU::RenderBundle::drawCount const):
(WebGPU::RenderBundle::lastError const):
* Source/WebGPU/WebGPU/RenderBundleEncoder.h:
(WebGPU::RenderBundleEncoder::createInvalid):
(WebGPU::RenderBundleEncoder::device const):
* Source/WebGPU/WebGPU/RenderBundleEncoder.mm:
(WebGPU::RenderBundleEncoder::returnIfEncodingIsFinished):
(WebGPU::Device::createRenderBundleEncoder):
(WebGPU::RenderBundleEncoder::RenderBundleEncoder):
(WebGPU::RenderBundleEncoder::makeInvalid):
(WebGPU::RenderBundleEncoder::addResource):
(WebGPU::RenderBundleEncoder::executePreDrawCommands):
(WebGPU::RenderBundleEncoder::draw):
(WebGPU::RenderBundleEncoder::maxVertexBufferIndex const):
(WebGPU::RenderBundleEncoder::maxBindGroupIndex const):
(WebGPU::RenderBundleEncoder::errorValidatingDraw const):
(WebGPU::RenderBundleEncoder::errorValidatingDrawIndexed const):
(WebGPU::RenderBundleEncoder::runIndexBufferValidation):
(WebGPU::RenderBundleEncoder::runVertexBufferValidation):
(WebGPU::RenderBundleEncoder::drawIndexed):
(WebGPU::RenderBundleEncoder::drawIndexedIndirect):
(WebGPU::RenderBundleEncoder::drawIndirect):
(WebGPU::RenderBundleEncoder::endCurrentICB):
(WebGPU::RenderBundleEncoder::validToEncodeCommand const):
(WebGPU::RenderBundleEncoder::finish):
(WebGPU::RenderBundleEncoder::replayCommands):
(WebGPU::RenderBundleEncoder::insertDebugMarker):
(WebGPU::RenderBundleEncoder::popDebugGroup):
(WebGPU::RenderBundleEncoder::pushDebugGroup):
(WebGPU::RenderBundleEncoder::setBindGroup):
(WebGPU::RenderBundleEncoder::setIndexBuffer):
(WebGPU::RenderBundleEncoder::recordCommand):
(WebGPU::RenderBundleEncoder::setPipeline):
(WebGPU::RenderBundleEncoder::setVertexBuffer):
(wgpuRenderBundleEncoderSetVertexBuffer):
* Source/WebGPU/WebGPU/RenderPassEncoder.h:
(WebGPU::RenderPassEncoder::create):
(WebGPU::RenderPassEncoder::createInvalid):
* Source/WebGPU/WebGPU/RenderPassEncoder.mm:
(WebGPU::RenderPassEncoder::RenderPassEncoder):
(WebGPU::RenderPassEncoder::~RenderPassEncoder):
(WebGPU::RenderPassEncoder::beginOcclusionQuery):
(WebGPU::RenderPassEncoder::endOcclusionQuery):
(WebGPU::RenderPassEncoder::addResourceToActiveResources):
(WebGPU::RenderPassEncoder::runIndexBufferValidation):
(WebGPU::RenderPassEncoder::runVertexBufferValidation):
(WebGPU::RenderPassEncoder::maxVertexBufferIndex const):
(WebGPU::RenderPassEncoder::maxBindGroupIndex const):
(WebGPU::RenderPassEncoder::errorValidatingAndBindingBuffers):
(WebGPU::RenderPassEncoder::errorValidatingDrawIndexed const):
(WebGPU::RenderPassEncoder::incrementDrawCount):
(WebGPU::RenderPassEncoder::issuedDrawCall const):
(WebGPU::RenderPassEncoder::executePreDrawCommands):
(WebGPU::RenderPassEncoder::draw):
(WebGPU::RenderPassEncoder::drawIndexed):
(WebGPU::RenderPassEncoder::drawIndexedIndirect):
(WebGPU::RenderPassEncoder::drawIndirect):
(WebGPU::RenderPassEncoder::endPass):
(WebGPU::RenderPassEncoder::executeBundles):
(WebGPU::RenderPassEncoder::insertDebugMarker):
(WebGPU::RenderPassEncoder::validatePopDebugGroup const):
(WebGPU::RenderPassEncoder::makeInvalid):
(WebGPU::RenderPassEncoder::popDebugGroup):
(WebGPU::RenderPassEncoder::setBindGroup):
(WebGPU::RenderPassEncoder::setBlendConstant):
(WebGPU::RenderPassEncoder::setIndexBuffer):
(WebGPU::RenderPassEncoder::setPipeline):
(WebGPU::RenderPassEncoder::setScissorRect):
(WebGPU::RenderPassEncoder::setStencilReference):
(WebGPU::RenderPassEncoder::setVertexBuffer):
(WebGPU::RenderPassEncoder::setViewport):
(wgpuRenderPassEncoderSetVertexBuffer):
(WebGPU::RenderPassEncoder::validateBindGroups): Deleted.
* Source/WebGPU/WebGPU/RenderPipeline.h:
(WebGPU::RenderPipeline::create):
(WebGPU::RenderPipeline::requiredBufferIndices const):
* Source/WebGPU/WebGPU/RenderPipeline.mm:
(WebGPU::createVertexDescriptor):
(WebGPU::Device::createRenderPipeline):
(WebGPU::RenderPipeline::RenderPipeline):
(WebGPU::RenderPipeline::primitiveTopology const):
(WebGPU::RenderPipeline::stripIndexFormat const):
* Source/WebGPU/WebGPU/ShaderModule.mm:
(WebGPU::handleShaderSuccessOrFailure):
* Source/WebGPU/WebGPU/Texture.h:
(WebGPU::Texture::isValid const): Deleted.
* Source/WebGPU/WebGPU/Texture.mm:
(WebGPU::Texture::destroy):
(WebGPU::Texture::validateImageCopyTexture):
(WebGPU::Texture::isValid const):
* Source/WebGPU/WebGPU/TextureView.h:
(WebGPU::TextureView::isValid const): Deleted.
* Source/WebGPU/WebGPU/TextureView.mm:
(WebGPU::TextureView::isValid const):
(WebGPU::TextureView::destroy):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderBundleEncoder.cpp:
(WebKit::RemoteRenderBundleEncoder::setVertexBuffer):
(WebKit::RemoteRenderBundleEncoder::unsetVertexBuffer):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderBundleEncoder.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderBundleEncoder.messages.in:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderPassEncoder.cpp:
(WebKit::RemoteRenderPassEncoder::setVertexBuffer):
(WebKit::RemoteRenderPassEncoder::unsetVertexBuffer):
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderPassEncoder.h:
* Source/WebKit/GPUProcess/graphics/WebGPU/RemoteRenderPassEncoder.messages.in:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDescriptor.cpp:
(WebKit::WebGPU::ConvertToBackingContext::convertToBacking):
(WebKit::WebGPU::ConvertFromBackingContext::convertFromBacking):
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDescriptor.h:
* Source/WebKit/Shared/WebGPU/WebGPURenderPassDescriptor.serialization.in:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderBundleEncoderProxy::setVertexBuffer):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderBundleEncoderProxy.h:
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.cpp:
(WebKit::WebGPU::RemoteRenderPassEncoderProxy::setVertexBuffer):
* Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteRenderPassEncoderProxy.h:

Canonical link: https://commits.webkit.org/273966@main
